### ISSUE TITLE
Facing issue with Proguard and Duktape

### ISSUE THREAD

**Issue by swapnilgt** [cashapp/quickjs-java#176]
I am using Duktape for evaluating JS in my android project. The version of duktape-android used is 0.9.6. When trying to evaluate a JS using the method:

`duktape.evaluate(script)`

the code runs fine without Proguard minification. But, when I run this piece of code with Proguard enabled, the app crashes with the following exception:

```
java.lang.RuntimeException: Unable to create application <package>.MyApplication: com.squareup.duktape.DuktapeException: TypeError: undefined not callable
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6857)
        at android.app.ActivityThread.access$1300(ActivityThread.java:268)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1982)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:7807)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1047)
     Caused by: com.squareup.duktape.DuktapeException: TypeError: undefined not callable
        at JavaScript.eval(?:2)
        at com.squareup.duktape.Duktape.evaluate(Native Method)
        at com.squareup.duktape.Duktape.a(:59)
        at com.squareup.duktape.Duktape.b(:67)
```
*****************************************************************
**I upgraded to latest version of Duktape (1.3.0) and now the exception message changed a bit**
It says:

```
java.lang.RuntimeException: Unable to create application in.zeta.apollo.acssdkandroid.MyApplication: com.squareup.duktape.DuktapeException: ReferenceError: identifier 'require' undefined
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6857)
        at android.app.ActivityThread.access$1300(ActivityThread.java:268)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1982)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:7807)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1047)
     Caused by: com.squareup.duktape.DuktapeException: ReferenceError: identifier 'require' undefined
        at JavaScript.eval(?:2)
        at com.squareup.duktape.Duktape.evaluate(Native Method)
        at com.squareup.duktape.Duktape.b(:66)
```

Can someone please help with this?

**Comment by JakeWharton** [comment]
I don't think Duktape supports a module system.

**Comment by JakeWharton** [comment]
I have published Duktape-Android 1.4.0 which bundles Duktape 2.6.0. This will be the final release of a Duktape repackaging as we are switching to QuickJS.
